src/GafferImage/ImageWriter.cpp

```diff
@@ -869,6 +935,12 @@ IECore::MurmurHash ImageWriter::hash( const Context *context ) const
 void ImageWriter::execute() const
 {
 	// Create an OIIO::ImageOutput
+	Context * c = const_cast<Context *>( Context::current() );
+	c->set( "gaffer:colorspace", defaultColorSpace()( currentFileFormat() ) );
```

Comment:
Let's call this something like  "__imageWriter:colorSpace", to indicate that it's a private implementation detail. At some point it might even be nice if we had a DeleteContextVariables node we could use to prevent it being visible upstream at all, but we can leave that for another time.